conversion/conversion.go

```diff
@@ -0,0 +1,670 @@
+package conversion
```

Comment:
This package duplicates about 90% of main.go.
I understand that we want to make these functions available via the web apis, but I don't think duplicating this code is a good way to achieve that.

It's going to be very hard to maintain and very confusing for anyone looking at this code.

Suggest that:
a) we move the code we need from main.go into a separate file that we access from main.go.
b) we put that file in internal (I don't think having a separate module is needed: much of the code calls into internal anyway)
c) call it internals/conversion.go